packages/cli/src/commands/pipelines/diff.ts

```diff
@@ -85,31 +85,41 @@ export default class PipelinesDiff extends Command {
 
   kolkrabbi: KolkrabbiAPI = new KolkrabbiAPI(this.config.userAgent, () => this.heroku.auth)
 
-  getAppInfo = async (appName: string, appId: string): Promise<AppInfo> => {
+  getAppInfo = async (appName: string, appId: string, generation: string): Promise<AppInfo> => {
     // Find GitHub connection for the app
     const githubApp = await this.kolkrabbi.getAppLink(appId)
       .catch(() => {
         return {name: appName, repo: null, hash: null}
       })
 
     // Find the commit hash of the latest release for this app
-    let slug: Heroku.Slug
+    let slug: Slug
+    let ociImages: Array<OciImage>
+    let commit: string | undefined
+
     try {
       const {body: releases} = await getReleases(this.heroku, appId)
       const release = releases.find(r => r.status === 'succeeded')
-      if (!release || !release.slug) {
+      if (!release || !(release.slug || release.oci_image)) {
         throw new Error(`no release found for ${appName}`)
       }
 
-      slug = await this.heroku.get<Heroku.Slug>(`/apps/${appId}/slugs/${release.slug.id}`, {
-        headers: {Accept: V3_HEADER},
-      }).then(res => res.body)
-    // tslint:disable-next-line: no-unused
+      if (generation === 'cedar' && release.slug) {
+        slug = await this.heroku.get<Slug>(`/apps/${appId}/slugs/${release.slug.id}`, {
+          headers: {Accept: SDK_HEADER},
+        }).then(res => res.body)
+        commit = slug.commit!
+      } else if (generation === 'fir' && release.oci_image) {
```

Comment:
It might if there's a succeeded release that doesn't have a `slug` (for Cedar apps) or `oci_image` (for Fir apps). I don't believe it would happen and, furthermore, `commit` will keep it's `undefined` value and be assigned to hash, just like the `catch` block does, which is what I'm intending: same behavior than error conditions.